### PR TITLE
Support escaping characters in keys

### PR DESCRIPTION
**Pull request by andy-s-clark**
Allow property keys such as:
`button.dict.fo\\o         = should have a backslash`
`button.dict.bar\.bar       = should have a dot`

### CODE REVIEW

File: example/tonic.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-var propertiesToObject = require('java.properties.js').default;
+var propertiesToObject = require('/home/andy.clark/work/java.properties.js').default;
```
Comment (willemdewit) [inline example/tonic.js]: This path is incorrect for runkit, it only resolves by module name.

**Review by willemdewit — CHANGES_REQUESTED** [review]
Hi, thank you for submitting a pull-request. I wonder what your use-case is. 
If the tonic example is fixed this is good to merge.

**Comment by andy-s-clark** [comment]
I have removed the change to example/tonic.js that I accidentally included.

In general it seems like a good idea to allow escaping of delimiting characters. The use case I ran into is for HTML verification meta tags. In this case the key is the name of the meta tag and the value is the content of the meta tag. For example [Bing's meta tag verification](http://blogtimenow.com/seo/meta-tag-verification-google-bing-alexa/):

    <meta name=”msvalidate.01″ content=”E6888vB9GDSGDSGJOJOJOIWF3r2BC613BBE7″ />
    <meta name=”google-site-verification″ content=”sdffdsfgsgsgdshernynyu-kyuk7ZgsdiCK1e-khkjhkhmbdbvc″ />

Written as properties:

    metatags.msvalidate\\.01 = E6888vB9GDSGDSGJOJOJOIWF3r2BC613BBE7
    metatags.google-site-verification = sdffdsfgsgsgdshernynyu-kyuk7ZgsdiCK1e-khkjhkhmbdbvc

**Review by willemdewit — APPROVED** [review]

**Comment by andy-s-clark** [comment]
Thanks for module and the quick merge!

**Pull request merged**